From: John W. Linville <linville@tuxdriver.com>
Date: Tue, 30 Mar 2010 13:47:39 -0400
Subject: [PATCH] ssb: avoid null ptr deref in ssb_is_sprom_available

[A later version of this is
 commit da1fdb02d9200ff28b6f3a380d21930335fe5429 upstream.]

Some older devices don't have chipcommon, but they do have SPROM.

Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 drivers/ssb/sprom.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/ssb/sprom.c b/drivers/ssb/sprom.c
index 55eb9b0..874d8f1 100644
--- a/drivers/ssb/sprom.c
+++ b/drivers/ssb/sprom.c
@@ -178,6 +178,10 @@ const struct ssb_sprom *ssb_get_fallback_sprom(void)
 
 bool ssb_is_sprom_available(struct ssb_bus *bus)
 {
+	/* some older devices don't have chipcommon, but they have sprom */
+	if (!bus->chipco.dev)
+		return true;
+
 	/* status register only exists on chipcomon rev >= 11 */
 	if (bus->chipco.dev->id.revision < 11)
 		return true;
-- 
1.6.2.5

